docs/Project-Configuration/Merge-Queue.md

```diff
@@ -117,4 +117,83 @@ branch protection rule for one or more "evergreen/variant_name" statuses. You
-tasks, but you still wish them to run automatically.
\ No newline at end of file
+tasks, but you still wish them to run automatically.
+
+**Q:** How can I turn off the merge queue to block users from merging?
+
+**A:** Check the "Lock branch" setting in the branch protection rules. There are
+*no changes to make on the Evergreen side.
+
+## Merge Queue Behavior
+
+The merge queue is not trying to merge individual PRs, but groups of PRs. This
+leads to some unintuitive behavior.
+
+Here is a typical sequence of events when a user adds 2 PRs to the merge queue.
+In this case we assume "Minimum pull requests to merge" is set to 1, "Maximum
+pull requests to merge" is set to 5, and "Maximum pull requests to build"
+("Build concurrency") is set to 5.
+
+The diagram names its branches like "main/pr-1", but in reality they are named
+like "gh-readonly-queue/main/pr-1-\<hash\>".
+
+```mermaid
+sequenceDiagram
+    User->>GitHub: Add PR 1 to merge queue
+    Note over GitHub: Create branch main/pr-1
+    GitHub->>Evergreen: Send merge_group webhook
+    Note over Evergreen: Create version for PR 1
+    Evergreen->>GitHub: Clone main/pr-1
+    User->>GitHub: Add PR 2 to merge queue
+    Note over GitHub: Create branch main/pr-2
+    GitHub->>Evergreen: Send merge_group webhook
+    Note over Evergreen: Create version for PR 1 + 2
+    Evergreen->>GitHub: Clone main/pr-2
+    Note over Evergreen: Tasks pass for PR 1
+    Evergreen->>GitHub: Send variant and version statuses for PR 1
+    Note over Evergreen: Tasks pass for PR 1 + 2
+    Evergreen->>GitHub: Send variant and version statuses for PR 1 + 2
+    Note over GitHub: Merge PR 1 + 2
+```
+
+There is some unintuitive behavior to be aware of:
+
+1. If main/pr-1 succeeds and main/pr-2 fails, only main/pr-1 will be merged.
+*But note that merging main/pr-1 waits for main/pr-2 to fail or for the timeout
+to be hit, because the "Maximum pull requests to merge" is set to 5. That is,
+GitHub is conceptualing the fundamental unit of work as the merge group, not the
+PR.*
+2. If main/pr-1 fails and main/pr-2 succeeds, GitHub will remove pr-1 from the
+queue and re-run main/pr-2 (not rebased on pr-1). *This is conceptually similar
+to the above. GitHub is trying to merge a group of PRs, not individual PRs.*
+3. If both succeed, main/pr-2 (rebased on pr-1) will be merged. *But merging
+main/pr-1 waits for main/pr-2 to succeed, because the maximum pull requests to
+merge is set to 5, and, again, GitHub is testing the entire group.*
+
+If, in the branch protection rules, "only merge non-failing pull requests" is
+checked, then the merge will not happen if any of the PRs fail. Otherwise, if
+main/pr-1 is red and main/pr-2 is green, then the latter will be merged, which
+contains both PRs.
+
+The temporary branch gets deleted only after the the PR is merged, or if the PR
+fails the check or is removed from the queue.
+
+The descriptions of the merge queue settings in the repo branch protection rules
```

Comment:
nit: this part seems a little out of place with the explanation of the merge queue behavior above, can we move it to a sub-section? Might also help to start off with a sentence explaining what we're talking about (i.e. comparisons between GitHub repo admin settings that aren't well-explained by GitHub).